proxylib/proxylib/connection.go

```diff
@@ -59,6 +62,117 @@ type Connection struct {
 	ReplyBuf   InjectBuf // Buffer for injected frames in reply direction
 }
 
+func NewConnection(instance *Instance, proto string, connectionId uint64, ingress bool, srcId, dstId uint32, srcAddr, dstAddr, policyName string, origBuf, replyBuf *[]byte) (FilterResult, *Connection) {
+	// Find the parser for the proto
+	parserFactory := GetParserFactory(proto)
+	if parserFactory == nil {
+		return UNKNOWN_PARSER, nil
```

Comment:
Why isn't `NewConnection` returning an error for these errors?

Reply:
Did not think of it, as the result is returned via the shared library interface as-is (not as a go error). However, it was simple enough for the `Result` type to implement the error interface so I changed that. Thanks!